packages/nimble-components/src/text-field/index.ts

```diff
@@ -51,7 +51,9 @@ export class TextField extends FoundationTextField implements ErrorPattern {
     public override connectedCallback(): void {
         super.connectedCallback();
         if (this.textContent?.trim()) {
-            this.style.height = `calc(${labelHeight.getValueFor(this)} + ${controlHeight.getValueFor(this)})`;
+            this.style.height = `calc(${labelHeight.getValueFor(
```

Comment:
This is definitely not how we should be styling controls. As much as possible we need to be using CSS for layout and this is straying farther from that. I haven't tried to evaluate what is going on here but we need strong justification for going down this path that I do not see discussed in the PR.

Reply:
This was the only way I could think of. I'm not aware of any way to do this in CSS. Do you think I am missing a better approach, or do you just want me to be more explicit in the description that this is the only way I could come up with?

Reply:
> I'm not aware of any way to do this in CSS.

The issue description in it's possible solutions gives a CSS-based approach. Requiring an additional class or attribute is not a great solution but if you tried and eliminated CSS based solutions such as that one some context of what you tried would be useful. 

> If someone programmatically adds/removes a label after the element is added to the DOM, the element will have the wrong height. I don't think we care very much to support that scenario.

I strongly disagree with this assessment. Our controls are designed to dynamically respond to attributes / properties / slotted content, etc. I would expect responsiveness to changes in configuration inline with our other controls.

Reply:
I talked with Milan and he turned me on to an approach using FAST's `slotted` directive which is already being used in the text field and number field templates. The idea is that the `slotted` directive will assign any slotted content (i.e. the label text) to a `defaultSlottedNodes` observable member. I can then write a `defaultSlottedNodesChanged()` method that sets (or removes) a `labeled` class on the `nimble-text-field` element. Then writing the CSS to have a different height based on the presence of the `labeled` class is trivial.

However, I am not seeing the `slotted` directive work the way it should. `defaultSlottedNodes` is never populated with my label text node. It is initialized to an empty array, and never changes after that.

Reply:
It looks like a super simple approach will work. If we just stop explicitly setting the height on the text/number field, then the height ends up being calculated correctly, whether there is a label (48px) or not (32px). I tested this out in Storybook.